shop_stripe/payment.py

```diff
@@ -52,6 +57,31 @@ def charge(self, cart, request):
             msg = "Stripe returned status '{status}' for id: {id}"
             raise stripe.error.InvalidRequestError(msg.format(**charge))
 
+    def subscribe(self, cart, request):
+        """
+        Create a new customer and subscribe the customer
+        to the default payment plan.
+        """
+        user_id = cart.customer.user_id;
+        user = User.objects.get(id=user_id)
+
+        try:
+            customer = StripeCustomer.objects.get(user_id=user_id)
+            stripe_customer = stripe.Customer.retrieve(customer.stripe_customer_id)
+        except ObjectDoesNotExist:
```

Comment:
Thanks for the feedback! I agree with you about not using the generic ObjectDoesNotExist, and instead I'll change it to expect StripeCustomer.DoesNotExist instead.

Regarding using try/except as control blocks. The original code had the same approach so I kept it, and I believe it is because it seems stripe.Customer doesn't have a get_or_create method. While testing, I noticed I was able to create the same customer twice given identical parameters.

In addition, stripe seems to be very limited when it comes to retrieving customers as well (can only query on stripe customer id).
Stripe API for reference: https://stripe.com/docs/api/python#create_customer